### PR TITLE
Allow -stage cherry picks, align tag format with blog

### PR DESCRIPTION
**Pull request by diox**
Fixes https://github.com/mozilla/extension-workshop/issues/1236

### CODE REVIEW

**Comment by diox** [comment]
We should change that regexp tbh... https://github.com/mozilla/extension-workshop/blob/da00311877d21bd88e129066ffdcb10ac9831701/.circleci/config.yml#L124

**Comment by diox** [comment]
Here is addons-blog for comparison:
https://github.com/mozilla/addons-blog/blob/40d3a1276ff5c379283abb593330ae7ed1dc35c9/.circleci/config.yml#L238

File: .circleci/config.yml [hunk 1]
```diff
@@ -158,13 +160,16 @@ workflows:
...
             - build
           filters:
             branches:
               ignore: /.*/
             tags:
-              # match tags such as "v1.1.1" or "v1.1.1-stage"
-              only: /^v?[0-9]+(\.[0-9]+)*(?:-stage)?$/
+              # match tags such as "2022.03.08" or "2022.03.08-1", with or
+              # without a -stage suffix.
+              only: /^20\d{2}\.\d{2}\.\d{2}(?:-\d+)?(?:-stage)?$/
```
Comment (eviljeff) [inline .circleci/config.yml:175]: I don't suppose there is a way in yaml to define this regex once in the file so we don't have to repeat it?

File: README.md [hunk 2]
```diff
@@ -339,13 +339,13 @@ The site is auto-deployed on commits to `master` to https://extensionworkshop-de
 
 Tags with a version ending in `-stage` will be deployed to https://extensionworkshop.allizom.org/. You can check the version on stage with [the stage version link](https://extensionworkshop.allizom.org/__version__).
 
-A good example tag for a stage deploy would be `v2.0.1-stage`.
+A good example tag for a stage deploy would be `2022.03.03-stage`.
 
 ### Production Deploys
 
-Tags of the format `vN.N.N` will be deployed to https://extensionworkshop.com/. You can check the version on production with [the production version link](https://extensionworkshop.com/__version__).
+Tags matching `^v?[0-9]+(\.[0-9]+)*$` regular expression will be deployed to https://extensionworkshop.com/. You can check the version on production with [the production version link](https://extensionworkshop.com/__version__).
```
Comment (eviljeff) [inline README.md:346]: stray `v` in there, I believe.

**Review by eviljeff — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (diox) [reply README.md:346]: Ah actually I should update this now that the CircleCI config no longer allows tags in the `vN.N.N` format

File: .circleci/config.yml (hunk 1 above)
Comment (diox) [reply .circleci/config.yml:175]: Not trivially as far as I'm aware

**Pull request merged**